### ISSUE TITLE
Anchor should not be required in InfoWindow

### ISSUE THREAD

**Issue by michaelsync** [JustFly1984/react-google-maps-api#94]
# Please provide an explanation

I just upgraded the @react-google-maps/api package (^1.2.2) . The anchor property is the required property for InfoWindow now. My working infowindows code doesn't work after the upgrade. I fixed it by setting anchor={null}  but the InfoWindows is not visible anymore. 

I am looking for an example. The official example (https://react-google-maps-api-docs.netlify.com/#infowindow) doesn't even use `anchor` so not sure if those examples are using the latest version. The link for InfoWIndow example is https://github.com/JustFly1984/react-google-maps-api/blob/17949d752e3f8c03a943dbdbf8778c8934a6d986/packages/react-google-maps-api/src/components/drawing/InfoWindow.md. It doesn't use the `anchor` which is a mandatory field. 

Can we make this `anchor` not mandatory? 

![image](https://user-images.githubusercontent.com/192911/56003714-2e0ae480-5d0b-11e9-83dd-3eb1f29b58d8.png)

## Your Environment

os: windows

node --version 

v10.15.1
react version

"react": "^16.8.4",
    "react-bootstrap": "^1.0.0-beta.6",
    "react-dom": "^16.8.4",
    "react-router": "^5.0.0",
    "react-router-dom": "^5.0.0",

@react-google-maps/api version
    "@react-google-maps/api": "^1.2.2",
    "@types/googlemaps": "^3.30.19",

## How is it behave?

Need to set the anchor. If I set anchor={null then InfoWindow doesn't even show anymore. 

## How should it behave correctly?

This code used to work with 1.1.0. 

```
function MapComponent () {  
    const [selectedCar, setSelectedCar] = React.useState<Cab>({} as Cab);
    const [cabs, setCabs] = React.useState<Cab[]>([]);  
  
    React.useEffect(() => {  
        //Call API and set the data to "cabs" state (setCabs)
    );
    
    //Timeer => call API every x seconds and set the data to "cabs" state (setCabs)
    
    const createMarker = ()=> {
       const markers:any = [];
       
       cabs.forEach((tx:Cab) => { 
             markers.push(
         <Marker key={tx.CarNumber}
                    position={{ lat: tx.Latitude, lng: tx.Longitude }}
                    label={carNumberAndDirection}
                    onClick={() => {                     
                                setSelectedCar(tx);            
                    }}
                    />);
          
           return markers;
    }   

    const showInfoWindows = () => {
        var tx:Cab = selectedCar;  
        return (<InfoWindow 
                  position={{ lat: tx.Latitude, lng: tx.Longitude }}
                 >
                  <div className="Popup">
                    <div id="title">{tx.CarNumber}</div> 
                  </div>
                </InfoWindow>);
    }
    
    return (
        <div>
            <GoogleMap ...>
                {createMarker()} 
                {(selectedCar.VehicleID > 0) && showInfoWindows(selectedCar)}
            <GoogleMap>
        <div>
     );
};
```

I got this example: https://codesandbox.io/s/9oqw8vqnro?fontsize=14

I upgraded the @react-google-maps/api package to 1.2.2. The code doesn't use 'anchor' and no error. 

But in my project, I got this error. 

![image](https://user-images.githubusercontent.com/192911/56003977-8f7f8300-5d0c-11e9-88ca-4f42eb26fda2.png)

**Comment by michaelsync** [comment]
I got this example: https://codesandbox.io/s/9oqw8vqnro?fontsize=14

I upgraded the @react-google-maps/api package to 1.2.2. The code doesn't use 'anchor' and no error. 

But in my project, I got this error. 

![image](https://user-images.githubusercontent.com/192911/56003977-8f7f8300-5d0c-11e9-88ca-4f42eb26fda2.png)

**Comment by uriklar** [comment]
I deleted your comment from the Contributors are welcome issue. No need for duplication. This issue is enough

**Comment by michaelsync** [comment]
Thanks, @uriklar . 

I am stuck with this issue in 1.2.1 and duplicate identifier issue in 1.2.1 or 1.2.0. 

I found a weird rendering issue in 1.1.1. 

I am really stuck at this moment.

**Comment by uriklar** [comment]
Yeah 1.1.1 has some issues. I'm working on fixing everything now, but in the meantime, it looks like you Typescript config is validating 3rd party libraries as well. Maybe you can try excluding node_modules in your tsconfig?
There's no real bug here, just typescript errors. I don't think ts errors from a 3rd party should be failing your build

**Comment by michaelsync** [comment]
@uriklar 

Here is my typescript config. I created my project based on this one https://github.com/Microsoft/TypeScript-React-Starter. It is already excluded the node_module in ts.config. but I am still getting the error. Can you know if I need to change something in tsconfig or somwhere.  

````
{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "build/dist",
    "module": "esnext",
    "target": "es5",
    "lib": ["es6", "dom"],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "src",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "importHelpers": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true  
  },
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.ts"
  ]
}
````

Here is my tslint.json.

```
{
  "extends": [],
  "defaultSeverity": "warning",
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts",
      "coverage/lcov-report/*.js"
    ]
  }
}

`````

If you can release the new version with fixes on yarn today or tomorrow, I am happpy to get that one. Thanks.

**Comment by uriklar** [comment]
Hi @michaelsync,
@JustFly1984 will release it probably sometime later today.

In the meantime you can build it yourself from the develop branch:
```
git clone -b develop git://github.com/JustFly1984/react-google-maps-api.git
cd react-google-maps-api/packages/react-google-maps-api
yarn install
yarn build
```

Then take the resulting ```lib``` folder and paste it inside your project:
```node_modules/@react-google-maps/api```

**Comment by michaelsync** [comment]
Hi @uriklar 

Thank you. I just upgraded to `1.2.3-alpha.1` but am still having the typescript issue. 

I think you have fixed the anchor issue in this file /src/components/drawing/InfoWindow.tsx. 

but haven't updated the **lib\components\drawing\InfoWindow.d.ts**.

I have forked the repo but I can't find this type file `InfoWindow.d.ts`.

**Comment by uriklar** [comment]
The Typescript error has to do with  conflicting versions of @types/react.
After upgrading to 1.2.3 try deleting node modules and yarn.lock (or npm lock file) and reinstalling. That should do the trick

**Comment by michaelsync** [comment]
@uriklar 

I tried that but it didn't work. I don't think it's about @types/react. How does '..node_modules\**@react-google-maps\api\lib\components\drawing\InfoWindow.d.ts**' get generated or updated? I know you updated '/src/components/drawing/InfoWindow.tsx' but not `lib\components\drawing\InfoWindow.d.ts`

![image](https://user-images.githubusercontent.com/192911/56071928-3af40a80-5dd5-11e9-8e1a-05e06b4b476c.png)

**Comment by uriklar** [comment]
Hi @michaelsync, you're right. Something got screwed up when publishing. Both in InfoWindow and in the @types dependency issue :-/
We'll get it sorted out.

**Comment by michaelsync** [comment]
@uriklar Thank you. Hope to get the release very soon. 

I can't manually put and copy the output to node_modules manually because we build the code on CI before the deployment. We excluded the whole node_modules folder from git.

**Comment by michaelsync** [comment]
@uriklar any update, guys? 

if the release with fix is not going to happen this week, please suggest me an alternative. I don't think I will get enough time to change the whole control to something else? I can manually build and fix it myself and put the output in a different folder (not in node_modules which I excluded it in git) and use it  ?

**Comment by uriklar** [comment]
Hi. @JustFly1984 has publishing access so you'll have to wait for him to be online. I'm sure it will be sometime today though (at least in my timezone 😀)

In the meantime you can build the package manually (from the branch of the currently open PR) as i suggested and either:
1. publish it on npm temporarily 
2. Push it to git ane point npm there
3. Find out how to point npm at a package inside a Lerna repo (because i couldn't find the way to do it)

But best bet is to fix it locally and wait for the release

**Comment by JustFly1984** [comment]
@michaelsync 
I've published 1.2.3-alpha.3

**Comment by michaelsync** [comment]
@uriklar @JustFly1984  Thanks, guys. I managed to upgrade the package and run the project. I will do some regression testings to see any side-effect after the upgrade. 

Thanks again, guys! You guys are awesome.
